x-pack/plugins/security_solution/public/management/pages/event_filters/view/components/event_filter_delete_modal.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useCallback, useEffect } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiText,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useDispatch } from 'react-redux';
+import { Dispatch } from 'redux';
+import { i18n } from '@kbn/i18n';
+import { useEventFiltersSelector } from '../hooks';
+import {
+  getDeleteError,
+  getItemToDelete,
+  isDeletionInProgress,
+  wasDeletionSuccessful,
+} from '../../store/selector';
+import { AppAction } from '../../../../../common/store/actions';
+import { useToasts } from '../../../../../common/lib/kibana';
+
+export const EventFilterDeleteModal = memo<{}>(() => {
+  const dispatch = useDispatch<Dispatch<AppAction>>();
+  const toasts = useToasts();
+
+  const isDeleting = useEventFiltersSelector(isDeletionInProgress);
+  const eventFilter = useEventFiltersSelector(getItemToDelete);
+  const wasDeleted = useEventFiltersSelector(wasDeletionSuccessful);
+  const deleteError = useEventFiltersSelector(getDeleteError);
+
+  const onCancel = useCallback(() => {
+    dispatch({ type: 'eventFilterDeletionReset' });
+  }, [dispatch]);
+
+  const onConfirm = useCallback(() => {
+    dispatch({ type: 'eventFilterDeleteSubmit' });
+  }, [dispatch]);
+
+  // Show toast for success
+  useEffect(() => {
+    if (wasDeleted) {
+      toasts.addSuccess(
```

Comment:
I think you can manage this with the `useEventFiltersNotification` hook adding the new conditions for delete action.

Reply:
I feel that having the notifications closer to the code that generates them makes the component (in this case the delete modal) clearer to understand. Someone looking at this component would not be able to understand how errors/success are presented back to the UI - they would have to go look elsewhere. I think it also overloads that hook in that it starts to have too broad of concerns. I remember having that same feedback when that Trusted Apps approach was done - several new approaches were trialed during that implementation, this being one of them.

Give this background, let me know if you still feel this should be moved over to the hook - I can do that if feels like the better approach.

Reply:
Oh I see. Yeah it make sense. 
Idea - Maybe we could move this logic to the middleware (not now) just to keep the component itself as clean as we can. What I'm thinking is have an entry on our redux store for notifications and have a component that listens it to display the toast or not. 
Doing this we could move all of this logic (this and the ones in the hook) to the middleware and let components more clean.

Reply:
@dasansol92 yeah, that could work. Maybe we add this as a topic of discussion for our weekly eng. office hours.